includes/admin/views/html-admin-page-addons.php

```diff
@@ -11,124 +11,148 @@
+							<?php echo esc_html( $section->label ); ?>
+						</a>
+					</li>
+				<?php endforeach; ?>
+			</ul>
+			<div id="marketplace-current-section-name" class="current-section-name"><?php echo esc_html( $current_section_name ); ?></div>
+		</div>
+		</div>
+
+	<div class="wp-header-end"></div>
+
+	<div class="wrap">
+		<div class="marketplace-content-wrapper">
+			<?php if ( isset( $_GET['search'] ) ) : // phpcs:ignore WordPress.Security.NonceVerification.Recommended ?>
+				<h1 class="search-form-title">
+					<?php // translators: search keyword. ?>
+					<?php printf( esc_html__( 'Search results for "%s"', 'woocommerce' ), esc_html( sanitize_text_field( wp_unslash( $_GET['search'] ) ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended ?>
+				</h1>
+			<?php endif; ?>
+
+			<?php if ( '_featured' === $current_section ) : ?>
+				<div class="addons-featured">
+					<?php
 					$featured = WC_Admin_Addons::get_featured();
-				?>
-			</div>
-		<?php endif; ?>
-		<?php if ( '_featured' !== $current_section && $addons ) : ?>
-			<?php if ( 'shipping_methods' === $current_section ) : ?>
-				<div class="addons-shipping-methods">
-					<?php WC_Admin_Addons::output_wcs_banner_block(); ?>
+					?>
 				</div>
 			<?php endif; ?>
-			<?php if ( 'payment-gateways' === $current_section ) : ?>
-				<div class="addons-shipping-methods">
-					<?php WC_Admin_Addons::output_wcpay_banner_block(); ?>
-				</div>
+			<?php if ( '_featured' !== $current_section && $addons ) : ?>
+				<?php if ( 'shipping_methods' === $current_section ) : ?>
+					<div class="addons-shipping-methods">
+						<?php WC_Admin_Addons::output_wcs_banner_block(); ?>
+					</div>
+				<?php endif; ?>
+				<?php if ( 'payment-gateways' === $current_section ) : ?>
+					<div class="addons-shipping-methods">
+						<?php WC_Admin_Addons::output_wcpay_banner_block(); ?>
+					</div>
+				<?php endif; ?>
+				<ul class="products">
+					<?php foreach ( $addons as $addon ) : ?>
+						<?php
+						if ( 'shipping_methods' === $current_section ) {
+							// Do not show USPS or Canada Post extensions for US and CA stores, respectively.
+							$country = WC()->countries->get_base_country();
+							if ( 'US' === $country
+								&& false !== strpos(
+									$addon->link,
+									'woocommerce.com/products/usps-shipping-method'
+								)
+							) {
+								continue;
+							}
+							if ( 'CA' === $country
+								&& false !== strpos(
+									$addon->link,
+									'woocommerce.com/products/canada-post-shipping-method'
+								)
+							) {
+								continue;
+							}
+						}
+						?>
+						<li class="product">
+							<div class="product-details">
+								<?php if ( ! empty( $addon->image ) ) : ?>
+									<span class="product-img-wrap"><img src="<?php echo esc_url( $addon->image ); ?>" /></span>
+								<?php endif; ?>
+								<a href="<?php echo esc_attr( WC_Admin_Addons::add_in_app_purchase_url_params( $addon->link ) ); ?>">
```

Comment:
this should `esc_url`

Reply:
https://github.com/woocommerce/woocommerce/pull/30391/commits/af9e2b17774d9d8f6a532d54c2e62cdf2bba466a